src/home/getters/get-github-user.ts

```diff
@@ -30,13 +27,25 @@ async function getNewSessionToken(): Promise<string | null> {
   const params = new URLSearchParams(hash.substr(1)); // remove the '#' and parse
   const providerToken = params.get("provider_token");
   if (!providerToken) {
-    return null;
+    const error = params.get("error_description");
+    // supabase auth provider has failed for some reason
+    console.error(`GitHub login provider: ${error}`);
   }
-  return providerToken;
+  return providerToken || null;
 }
 
-async function getNewGitHubUser(providerToken: string): Promise<GitHubUser> {
+async function getNewGitHubUser(providerToken: string | null): Promise<GitHubUser | null> {
   const octokit = new Octokit({ auth: providerToken });
-  const response = (await octokit.request("GET /user")) as GitHubUserResponse;
-  return response.data;
+
+  try {
+    const response = (await octokit.request("GET /user")) as GitHubUserResponse;
+    return response.data;
+  } catch (error: unknown) {
+    if (error.status === 403) {
```

Comment:
This displays a TypeScript error as `error` if `unknown`

Reply:
I stopped ts from typing catch() as `unknown` by default allowing the use of `any`.

`unknown` is safer but if we are going to have to cast it anyway to stop the error and we know what we are receiving then `any` makes as much sense to use

Reply:
While I am not against the idea, I think it is better to check the type with `instanceof` and use the Octokit provided [error library](https://github.com/octokit/request-error.js?tab=readme-ov-file#usage-with-octokit) to avoid recreating an error within the catch blocks

Reply:
implemented